lib/logger.js

```diff
@@ -112,8 +111,37 @@ class Logger {
             this._traceLogger = this._logger.child({
                 level: bunyan.TRACE
             });
-            // Set up handlers for uncaught extensions
-            this._setupRootHandlers();
+
+            // Avoid recursion if there are bugs in the logging code.
+            this.inLogger = false;
+
+            this.logUnhandledException = (err) => {
```

Comment:
This should be moved up to the `base_service` as well. The idea is that we are using the logger here, but doing something with unhandled exceptions is not the task of a logger itself.

Also, the `inLogger` recusrion protection was lost for `unhandledRejection` handler.

Reply:
I'll take another look. The logUnhandledException made sense to me in the logger module

Reply:
Moved to base_service.js